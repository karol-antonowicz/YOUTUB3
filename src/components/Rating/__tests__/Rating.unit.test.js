import React from 'react';
import {shallow} from 'enzyme';
import {Rating} from '../Rating'

describe('Rating', ()=> {
    test('renders Rating without props', ()=> {
        const wrapper = shallow(<Rating/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders Rating with likeCount', ()=> {
        const wrapper = shallow(<Rating likeCount={20}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders Rating with dislikeCount', ()=> {
        const wrapper = shallow(<Rating dislikeCount={20}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('renders Rating with likeCount and dislikeCount', ()=> {
        const wrapper = shallow(<Rating likeCount={1} dislikeCount={20}/>);
        expect(wrapper).toMatchSnapshot();
    });
})