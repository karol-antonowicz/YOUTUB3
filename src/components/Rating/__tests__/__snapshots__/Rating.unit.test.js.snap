// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders Rating with dislikeCount 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      20
    </span>
  </div>
</div>
`;

exports[`Rating renders Rating with likeCount 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      20
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span />
  </div>
</div>
`;

exports[`Rating renders Rating with likeCount and dislikeCount 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      1
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      20
    </span>
  </div>
  <Progress
    className="progress"
    percent={4.761904761904762}
    size="tiny"
  />
</div>
`;

exports[`Rating renders Rating without props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span />
  </div>
</div>
`;
